src/System.Buffers/src/System/Buffers/NativeBufferPool.cs

```diff
@@ -8,12 +8,31 @@
 
 namespace System.Buffers
 {
-    public unsafe sealed class NativeBufferPool<T> where T : struct
+    public unsafe sealed class NativeBufferPool<T> : IDisposable where T : struct
     {
         private const int NUMBER_OF_BUFFERS_IN_BUCKET = 50;
         private int _maxElements;
         private NativeBufferBucket<T>[] _buckets;
 
+        private static NativeBufferPool<byte> _sharedInstance = null;
+
+        public static NativeBufferPool<byte> SharedByteBufferPool
+        {
```

Comment:
The 2MB buffer allocation *only* happens if the Shared pool gets a request for a 2MB buffer; 2MB is the *max* bucket size it will allocate. The initial construction of a Pool does not allocate any bucket or memory, therefore using any locking mechanism will be much heavier and go against the lock-free pattern we are trying to achieve